File: app/src/main/java/nic/goi/aarogyasetu/utility/ExecutorHelper.java
/*
 * Copyright 2020 Government of India
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

package nic.goi.aarogyasetu.utility;

import java.util.concurrent.LinkedBlockingQueue;
import java.util.concurrent.ThreadPoolExecutor;
import java.util.concurrent.TimeUnit;

/**
 * Created by Kshitij Khatri on 21/03/20.
 */
public class ExecutorHelper {

    private static int numCores = Runtime.getRuntime().availableProcessors();

    public static ThreadPoolExecutor getThreadPoolExecutor() {
        return new ThreadPoolExecutor(numCores * 2, numCores * 2,
                60L, TimeUnit.SECONDS, new LinkedBlockingQueue<Runnable>());
    }
}

File: app/src/main/java/nic/goi/aarogyasetu/utility/ExecutorHelper.kt
package nic.goi.aarogyasetu.utility

import java.util.concurrent.LinkedBlockingQueue
import java.util.concurrent.ThreadPoolExecutor
import java.util.concurrent.TimeUnit

/**
 * Created by Kshitij Khatri on 21/03/20.
 */
object ExecutorHelper {

    private val numCores = Runtime.getRuntime().availableProcessors()

    val threadPoolExecutor: ThreadPoolExecutor
        get() = ThreadPoolExecutor(
            numCores * 2, numCores * 2,
            60L, TimeUnit.SECONDS, LinkedBlockingQueue<Runnable>()
        )
}
